app/scenes/account_notifications/account_notifications.js

```diff
@@ -0,0 +1,515 @@
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import EventEmitter from 'service/utils/event_emitter';
+import {Preferences, RequestStatus} from 'service/constants';
+import {getPreferencesByCategory} from 'service/utils/preference_utils';
+
+import Section from './section';
+import SectionItem from './section_item';
+import SaveNotifcationsButton from './save_notifications_button';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return StyleSheet.create({
+        input: {
+            color: theme.centerChannelColor,
+            fontSize: 12,
+            height: 40
+        },
+        separator: {
+            height: 1,
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            marginHorizontal: 15
+        },
+        scrollView: {
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.03)
+        },
+        scrollViewContent: {
+            paddingBottom: 30
+        },
+        wrapper: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg
+        }
+    });
+});
+
+export default class AccountNofications extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            goBack: PropTypes.func.isRequired,
+            handleUpdateUserNotifyProps: PropTypes.func.isRequired
+        }),
+        config: PropTypes.object.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        myPreferences: PropTypes.object.isRequired,
+        saveRequestStatus: PropTypes.string.isRequired,
+        subscribeToHeaderEvent: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        unsubscribeFromHeaderEvent: PropTypes.func.isRequired
+    };
+
+    static navigationProps = {
+        renderRightComponent: (props, emitter) => {
+            return <SaveNotifcationsButton emitter={emitter}/>;
+        }
+    }
+
+    constructor(props) {
+        super(props);
+
+        const {currentUser} = props;
+        const notifyProps = currentUser.notify_props || {};
+        this.setStateFromNotifyProps(notifyProps);
+    }
+
+    componentWillMount() {
+        this.props.subscribeToHeaderEvent('save_notify_props', this.saveUserNotifyProps);
+    }
+
+    componentWillUnmount() {
+        this.props.unsubscribeFromHeaderEvent('save_notify_props');
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.currentUser !== this.props.currentUser) {
+            const {notify_props: notifyProps} = nextProps.currentUser;
+            this.setStateFromNotifyProps(notifyProps);
+        }
+        if (nextProps.saveRequestStatus === RequestStatus.SUCCESS && this.props.saveRequestStatus === RequestStatus.STARTED) {
+            this.props.actions.goBack();
+        }
+        if (nextProps.saveRequestStatus === RequestStatus.FAILURE && this.props.saveRequestStatus === RequestStatus.STARTED) {
```

Comment:
For ```nextProps.saveRequestStatus === RequestStatus.SUCCESS && this.props.saveRequestStatus === RequestStatus.STARTED``` we have to make sure we that the previous request status was```started``` or everytime that we navigate to the scene after a save the user will be immediately popped back to the previous scene because ```nextProps.saveRequestStatus``` and ```this.props.saveRequestStatus``` will  still be ```success``` since it doesn't get changed until a new request is fired.

Same idea for ```nextProps.saveRequestStatus === RequestStatus.FAILURE && this.props.saveRequestStatus === RequestStatus.STARTED```. We want to make sure that a new request actually failed before resetting the loading state for the button and resetting the user notify props.